include/jwt-cpp/jwt.h

```diff
@@ -3559,11 +3418,464 @@ namespace jwt {
+		 */
+		verifier& with_subject(const typename json_traits::string_type& sub) {
+			return with_claim("sub", basic_claim_t(sub));
+		}
+		/**
+		 * Set an audience to check for.
+		 * If any of the specified audiences is not present in the token the check fails.
+		 * \param aud Audience to check for.
+		 * \return *this to allow chaining
+		 */
+		verifier& with_audience(const typename basic_claim_t::set_t& aud) {
+			claims["aud"] = verify_ops::is_subset_claim<json_traits>{aud};
+			return *this;
+		}
+		/**
+		 * Set an audience to check for.
+		 * If the specified audiences is not present in the token the check fails.
+		 * \param aud Audience to check for.
+		 * \return *this to allow chaining
+		 */
+		verifier& with_audience(const typename json_traits::string_type& aud) {
+			typename basic_claim_t::set_t s;
+			s.insert(aud);
+			return with_audience(s);
+		}
+		/**
+		 * Set an id to check for.
+		 * Check is casesensitive.
+		 * \param id ID to check for.
+		 * \return *this to allow chaining
+		 */
+		verifier& with_id(const typename json_traits::string_type& id) { return with_claim("jti", basic_claim_t(id)); }
+
+		/**
+		 * Specify a claim to check for using the specified operation.
+		 * \param name Name of the claim to check for
+		 * \param fn Function to use for verifying the claim
+		 * \return *this to allow chaining
+		 */
+		verifier& with_claim(const typename json_traits::string_type& name, verify_check_fn_t fn) {
+			claims[name] = fn;
+			return *this;
+		}
+
+		/**
+		 * Specify a claim to check for equality (both type & value).
+		 * \param name Name of the claim to check for
+		 * \param c Claim to check for
+		 * \return *this to allow chaining
+		 */
+		verifier& with_claim(const typename json_traits::string_type& name, basic_claim_t c) {
+			return with_claim(name, verify_ops::equals_claim<json_traits>{c});
+		}
+
+		/**
+		 * Add an algorithm available for checking.
+		 * \param alg Algorithm to allow
+		 * \return *this to allow chaining
+		 */
+		template<typename Algorithm>
+		verifier& allow_algorithm(Algorithm alg) {
+			algs[alg.name()] = std::make_shared<algo<Algorithm>>(alg);
+			return *this;
+		}
+
+		verifier& allow_key(const jwt::jwk<json_traits>& key) {
+			std::string keyid = "";
+			if (key.has_key_id()) {
+				keyid = key.get_key_id();
+				auto it = keys.find(keyid);
+				if (it == keys.end()) { keys[keyid] = key_list(); }
+			}
+
+			keys[keyid].push_back(key);
+			return *this;
+		}
+
+		/**
+		 * Verify the given token.
+		 * \param jwt Token to check
+		 * \throw token_verification_exception Verification failed
+		 */
+		void verify(const decoded_jwt<json_traits>& jwt) const {
+			std::error_code ec;
+			verify(jwt, ec);
+			error::throw_if_error(ec);
+		}
+		/**
+		 * Verify the given token.
+		 * \param jwt Token to check
+		 * \param ec error_code filled with details on error
+		 */
+		void verify(const decoded_jwt<json_traits>& jwt, std::error_code& ec) const {
```

Comment:
A lot of code moved so the diff is very confusing. It hard to tell what is new 🙊